### PR TITLE
Convert MARC Generator to use Integration Configuration (PP-498)

### PR DESCRIPTION
**Pull request by jonathangreen**
## Description

🚨 This PR is based on: https://github.com/ThePalaceProject/circulation/pull/1539. That one will need to go in before this one does.

Converts the MARCExporter to be configured via the new config settings mechanism before starting work on PP-59 to make the MARCExporter configurable per collection.

## Motivation and Context

Continued work on the PP-4 epic.

## How Has This Been Tested?

Running test suite in CI.

## Checklist

- [X] I have updated the documentation accordingly.
- [X] All new and existing tests passed.

### CODE REVIEW

File: tests/api/admin/controller/test_patron_auth.py
[the remark this answers is not in the document]
Comment (jonathangreen) [reply tests/api/admin/controller/test_patron_auth.py:1]: Broke this out into this PR:
https://github.com/ThePalaceProject/circulation/pull/1539

File: api/admin/problem_details.py [hunk 1]
```diff
@@ -200,11 +193,11 @@
...
-MISSING_PATRON_AUTH_NAME = pd(
-    "http://librarysimplified.org/terms/problem/missing-patron-auth-name",
+MISSING_SERVICE_NAME = pd(
+    "http://librarysimplified.org/terms/problem/missing-service-name",
     status_code=400,
-    title=_("Missing patron auth service name."),
-    detail=_("You must identify the patron auth service by its name."),
+    title=_("Missing service name."),
+    detail=_("You must identify the service by its name."),
 )
```
[the remark this answers is not in the document]
Comment (jonathangreen) [reply api/admin/problem_details.py]: Broke this out to this PR: https://github.com/ThePalaceProject/circulation/pull/1539

File: tests/api/admin/controller/test_patron_auth.py
[the remark this answers is not in the document]
Comment (jonathangreen) [reply tests/api/admin/controller/test_patron_auth.py:1]: Broke this out to this PR: https://github.com/ThePalaceProject/circulation/pull/1539

File: tests/api/admin/controller/test_collections.py
[the remark this answers is not in the document]
Comment (jonathangreen) [reply tests/api/admin/controller/test_collections.py:1]: Broke this out to this PR: https://github.com/ThePalaceProject/circulation/pull/1539

File: core/marc.py [hunk 2]
```diff
@@ -481,118 +504,131 @@ class MARCExporterFacets(BaseFacets):
...
         filter.order = self.SORT_ORDER_TO_OPENSEARCH_FIELD_NAME[self.ORDER_LAST_UPDATE]
         filter.order_ascending = True
         filter.updated_after = self.start_time
 
 
-class MARCExporter:
-    """Turn a work into a record for a MARC file."""
-
-    NAME = ExternalIntegration.MARC_EXPORT
+class PositiveInt(ConstrainedInt):
```
[the remark this answers is not in the document]
Comment (tdilauro) [reply core/marc.py:516]: Actually, it looks like Pydantic has this already...
```
from pydantic import NonNegativeInt
```

[the remark this answers is not in the document]
Comment (jonathangreen) [reply core/marc.py]: Okay. I updated to use that.

**Comment by jonathangreen** [comment]
Updated PR based on code review feedback if you want to take another look @tdilauro

**Pull request merged**